### PR TITLE
floating tags

### PR DESCRIPTION
**Pull request by rcerven**
* OSBS-7169

Signed-off-by: Robert Cerven <rcerven@redhat.com>

### CODE REVIEW

**Comment by rcerven** [comment]
unit tests TODO

**Comment by rcerven** [comment]
unit tests added

File: atomic_reactor/plugins/post_import_image.py [hunk 1]
```diff
@@ -72,9 +72,9 @@ def run(self):
             self.log.info("Not importing failed build")
             return
 
-        self.primary_images = get_primary_images(self.workflow)
-        if not self.primary_images:
-            raise RuntimeError('Could not find primary images in workflow')
+        _, self.floating_images = get_primary_and_floating_images(self.workflow)
+        if not self.floating_images:
+            raise RuntimeError('Could not find floating images in workflow')
```
Comment (mlangsdorf) [inline atomic_reactor/plugins/post_import_image.py]: missing coverage

File: atomic_reactor/plugins/post_tag_from_config.py [hunk 2]
```diff
@@ -63,13 +64,29 @@ def parse_and_add_tags(self):
                 self.log.debug('Using additional unique tag %s', tag)
                 self.workflow.tag_conf.add_unique_image(tag)
 
+        for tag_suffix in self.tag_suffixes.get('floating', []):
```
Comment (mlangsdorf) [inline atomic_reactor/plugins/post_tag_from_config.py]: missing coverage for this entire block

**Comment by rcerven** [comment]
coverage fixed

**Comment by MartinBasti** [comment]
Could we have `get_primary_images` and `get_floating_images` as 2 separated functions?

I see only usage like:
```python
_, floating = get_primary_and_floating_images(...)
primary, _ = get_primary_and_floating_images(...)
```


Or you can have function `floating = filter_images(.., unique=False, floating=True, primary=False)`

**Comment by rcerven** [comment]
@MartinBasti  get_primary_and_floating_images was done according design

@twaugh  thoughts?

**Comment by twaugh** [comment]
Let's split it up into two methods, if that's how it's really used.

**Comment by rcerven** [comment]
ok will do

**Comment by rcerven** [comment]
I've split it into 2 methods

**Comment by mlangsdorf** [comment]
LGTM.

**Comment by twaugh** [comment]
Release notes draft updated:
- Better internal separation of floating tags and primary tags

**Pull request merged**